django/forms/models.py

```diff
@@ -1177,7 +1177,13 @@ def add_fields(self, form, index):
                 to_field = self.instance._meta.get_field(kwargs["to_field"])
             else:
                 to_field = self.instance._meta.pk
-            if to_field.has_default():
+
+            # Checking to_field is primary key field or form have data or not, if
+            # form have data then do not reset to_field default value.
+            if to_field.has_default() and (
+                to_field.attname == self.fk.remote_field.model._meta.pk.name
+                or not form.data
+            ):
```

Comment:
I see no test failures when removing `to_field.attname == self.fk.remote_field.model._meta.pk.name`, is it needed? If so, please add a test that covers this condition.

Reply:
I think we should add assertions to the `test_inlineformset_factory_ignores_default_pks_on_submit` :thinking: 
```diff
diff --git a/tests/model_formsets/test_uuid.py b/tests/model_formsets/test_uuid.py
index 2097bde82b..2084fc2987 100644
--- a/tests/model_formsets/test_uuid.py
+++ b/tests/model_formsets/test_uuid.py
@@ -43,6 +43,8 @@ class InlineFormsetTests(TestCase):
             }
         )
         self.assertTrue(formset.is_valid())
+        self.assertIsNone(formset.instance.uuid)
+        self.assertIsNone(formset.forms[0].instance.parent_id)
 
     def test_inlineformset_factory_nulls_default_pks_uuid_parent_auto_child(self):
```
If so, please do this in a separate PR/commit.

Reply:
> I see no test failures when removing `to_field.attname == self.fk.remote_field.model._meta.pk.name`, is it needed? If so, please add a test that covers this condition.

Yes, It's needed for validate condition in normal case, so Added new test case related to this in `test_inlineformset_factory_ignores_default_pks_set_uuid_none`

Reply:
A new test is not required as it is already covered by `test_inlineformset_factory_ignores_default_pks_on_submit`, only these two assertions are missing.